source/extensions/http/stateful_session/header/header.cc

```diff
@@ -1,29 +1,80 @@
 #include "source/extensions/http/stateful_session/header/header.h"
 
+#include "absl/container/inlined_vector.h"
+
 namespace Envoy {
 namespace Extensions {
 namespace Http {
 namespace StatefulSession {
 namespace Header {
 
+constexpr absl::string_view OriginPartFlag = ";origin:";
+
 void HeaderBasedSessionStateFactory::SessionStateImpl::onUpdate(
-    const Upstream::HostDescription& host, Envoy::Http::ResponseHeaderMap& headers) {
-  absl::string_view host_address = host.address()->asStringView();
+    absl::string_view host_address, Envoy::Http::ResponseHeaderMap& headers) {
+
+  if (factory_.mode_ == HeaderBasedSessionStateProto::PACKAGES) {
+    auto origin_header = headers.get(factory_.name_);
+    if (origin_header.size() != 1) {
+      ENVOY_LOG(debug, "Header {} not exist or occurs multiple times", factory_.name_);
+      return;
+    }
+
+    const std::string new_header =
+        absl::StrCat(host_address, OriginPartFlag, origin_header[0]->value().getStringView());
```

Comment:
It looks like the result will be `<HOST>;origin:<SESSION>`. Is it reversed?

Reply:
Yeah. The upstream address would be the first part to keep backward compatibility with previous format. This is also to keep same with https://github.com/grpc/proposal/blob/master/A60-xds-stateful-session-affinity-weighted-clusters.md

Reply:
Yes, I thought it would be a append. But reviewing your sequence diagram, it is a prepend, which makes sense.